[ONNX] Move graph transform functions to 'passes'

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #95651
* #95935
* #95929
* __->__ #95664
* #95676
* #95686
* #95650

This PR only moved code to their new location. No other actual code changes.

BowenBao committed:
[ONNX] Move graph transform functions to 'passes'

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Move graph transform functions to 'passes'"




[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Move graph transform functions to 'passes'"




[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Move graph transform functions to 'passes'"




[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Move graph transform functions to 'passes'"




[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Move graph transform functions to 'passes'"




[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Move graph transform functions to 'passes'"




[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Move graph transform functions to 'passes'"


This PR only moved code to their new location. No other actual code changes.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Move graph transform functions to 'passes'"


This PR only moved code to their new location. No other actual code changes.

[ghstack-poisoned]

wschin commented on torch/onnx/_internal/fx/passes/fx_to_onnxscript.py:
```diff
@@ -0,0 +1,510 @@
+    torch_args: tuple,
+    torch_kwargs: dict,
+):
+    """Validate the op between ONNX Runtime and PyTorch."""
```
What does this function really do? The doc string is basically the same as its name.

BowenBao replied:
This PR only splits code from one file to several.

titaiwangms replied:
This function is trying to validate op-by-op results with runtime shape generated inputs to provide op-level debugging. I have a plan to refactor this once we have settled the shape inference of graph (Currently, we don't have shape in proto and I haven't figured out how dynamic axes would affect the shape usage in this function, but this is default off).

wschin commented on torch/onnx/_internal/fx/passes/fx_to_onnxscript.py:
```diff
@@ -0,0 +1,510 @@
+            kwargs_onnx = _filter_incompatible_and_dtype_convert_kwargs(torch_kwargs)
+            ort_outputs = symbolic_fn(*input_onnx, **kwargs_onnx)
+
+            # TODO: add pytree structure comparison.
```
Impossible to compare pytree structure. FX allows recursive collections such as `List[Dict[List[.....]`.

BowenBao replied:
Could you elaborate what does `recursive collections` mean? Is it just nested containers, or are you suggesting it's possible to have cycles?

And could you elaborate why such structure makes it impossible to compare?

titaiwangms replied:
Reopen the function related issue that we could have more talk about this: https://github.com/microsoft/onnx-script/issues/393

BowenBao committed:
Update on "[ONNX] Move graph transform functions to 'passes'"


This PR only moved code to their new location. No other actual code changes.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Move graph transform functions to 'passes'"


This PR only moved code to their new location. No other actual code changes.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Move graph transform functions to 'passes'"


This PR only moved code to their new location. No other actual code changes.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Move graph transform functions to 'passes'"


This PR only moved code to their new location. No other actual code changes.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Move graph transform functions to 'passes'"


This PR only moved code to their new location. No other actual code changes.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Move graph transform functions to 'passes'"


This PR only moved code to their new location. No other actual code changes.

[ghstack-poisoned]

BowenBao committed:
bump onnxscript on "[ONNX] Move graph transform functions to 'passes'"


This PR only moved code to their new location. No other actual code changes.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Move graph transform functions to 'passes'"


This PR only moved code to their new location. No other actual code changes.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Move graph transform functions to 'passes'"


This PR only moved code to their new location. No other actual code changes.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Move graph transform functions to 'passes'"


This PR only moved code to their new location. No other actual code changes.

[ghstack-poisoned]
